File: Schedule/Schedule.Core/Models/Time.cs
namespace Schedule.Core.Models;

public class Time
{
    public int TimeId { get; set; }

    public TimeOnly Start { get; set; }

    public TimeOnly End { get; set; }

    public int Duration { get; set; }

    public int LessonNumber { get; set; }

    public int TypeId { get; set; }

    public bool IsDeleted { get; set; }

    public virtual ICollection<LessonChange> LessonChanges { get; set; } = new List<LessonChange>();

    public virtual ICollection<Lesson> Lessons { get; set; } = new List<Lesson>();

    public virtual TimeType Type { get; set; } = null!;
}

File: schedule.ios.applicaion/schedule.ios.applicaion/Models/Time.swift
//
//  Time.swift
//  schedule.ios.application
//
//  Created by Иван Светлов on 11.06.2023.
//

import Foundation

struct Time: Identifiable, Codable, Hashable, Equatable {
    var id: Int
    var start: String
    var end: String
    var duration: Int
    var lessonNumber: Int
    var type: TimeType
    var isDeleted: Bool
    
    func getRange() -> String {
        return "\(start)-\(end)"
    }
}
